pkg/util/log/log.go

```diff
@@ -40,20 +40,21 @@ var (
 	logsBuffer           = []func(){}
 	bufferLogsBeforeInit = true
 	bufferMutex          sync.Mutex
-	defaultStackDepth    = 3
+	defaultStackDepth    = 4
 )
 
 // DatadogLogger wrapper structure for seelog
 type DatadogLogger struct {
-	inner seelog.LoggerInterface
-	level seelog.LogLevel
-	extra map[string]seelog.LoggerInterface
-	l     sync.RWMutex
+	inner   seelog.LoggerInterface
+	verbose seelog.LoggerInterface
+	level   seelog.LogLevel
+	extra   map[string]seelog.LoggerInterface
+	l       sync.RWMutex
 }
 
 // SetupLogger setup agent wide logger
-func SetupLogger(i seelog.LoggerInterface, level string) {
-	Logger = setupCommonLogger(i, level)
+func SetupLogger(i seelog.LoggerInterface, level string, verboseLogger seelog.LoggerInterface) {
```

Comment:
What do you think of configuring the "verbose" logger similar to how the "extra" loggers are added? Ie, implement a method similar to [`registerAdditionalLogger`](https://github.com/DataDog/datadog-agent/blob/fb524aa1edb858348329ee78dc16ca9b7291d361/pkg/util/log/log.go#L139)

This would allow very minimal changes to the public API (and hopefully avoid a diff with 44 files)

Reply:
After some consideration, I think it would make the most sense to leave it the way I've changed it (so the `SetupLogger` function has the `verboseLogging` boolean parameter). While adding a separate function (for example `SetupVerboseLogging`) would allow the current `SetupLogging` function signature to remain the same (and not result in changing 44 files), I think it makes the most sense to tie the verbose logging to the `SetupLogger` function, since anything that wants verbose logging would also want normal logging. Put differently, you would never want to call a a hypothetical `SetupVerboseLogger` function without also calling a `SetupLogger` function first.